### ISSUE TITLE
In version 1.5.0 the Content-type header is application/json instead of multipart/form-data

### ISSUE THREAD

**Issue by dawidreedsy** [axios/axios#5842]
### Describe the bug

In version 1.4.0 of axios when sending the form data object the content type is properly set.
```
    POST /api/images HTTP/1.1\r\n
    Accept: application/json, text/plain, */*\r\n
    Content-Type: multipart/form-data; boundary=--------------------------908413013073488028571171\r\n
    Authorization: Bearer TEST\r\n\
    User-Agent: axios/1.4.0\r\n
    Content-Length: 117151\r\n
    Accept-Encoding: gzip, compress, deflate, br\r\n
    Host: localhost:8200\r\n
    Connection: keep-alive\r\n
```
When using 1.5.0 though i got this HTTP request
```
    POST /api/images HTTP/1.1\r\n
    Accept: application/json, text/plain, */*\r\n
    Content-Type: application/json\r\n
    Authorization: Bearer TEST\r\n
    User-Agent: axios/1.5.0\r\n
    Content-Length: 4\r\n
    Accept-Encoding: gzip, compress, deflate, br\r\n
    Host: localhost:8200\r\n
    Connection: keep-alive\r\n
```
 
 I call the axios like this
 ```typescript
        const httpAgent = new http.Agent({
          keepAlive: true,
          keepAliveMsecs: 1000,
          maxSockets: 256
        });
       
      const instance = axios.create({
        baseURL: `http://localhost:8200`,
        httpAgent,
      });
    
      instance.defaults.headers.common['Content-Type'] = 'application/json';
      instance.defaults.headers.common['Authorization'] = `Bearer TEST`;
      instance.defaults.maxBodyLength = Infinity;
      instance.defaults.maxContentLength = Infinity;
      
       const payload = new FormData();
       const imageBuffer = fs.readFileSync('path to file')
      payload.append('imageFile', imageBuffer, {filename: 'test.png'});
      const response = await axios.post(`images`, payload);
 ```
 

I think it might be, because of this line:
 ```typescript
       instance.defaults.headers.common['Content-Type'] = 'application/json';
 ```
 if the line above is wrong I think the change should be marked as breaking or if the line is fine, then is should be a bug that should be fixed.
 

### Axios Version

1.5.0

### Node.js Version

18.10.0

**Comment by DigitalBrainJS** [comment]
This is not actually a new behavior, but it results from a bug fix with the merging of `Content-Type` in common headers. The feature it applies to was released with the major semver tag, but the `Content-Type` set by the common headers was simply ignored and the bug went unnoticed for a long time until #5825 was reported. Now some users will consider this a bug fix, others - breaking change. Thus it's quite a difficult question how to do it in a way that is acceptable to both parties. Fixing this issue essentially means reverting the fix for bug #5825.

**Comment by dawidreedsy** [comment]
Ok that make sense 👍
